### PR TITLE
tikv-ctl: allow get_all_regions from remote, drop-unapplied-raftlog

### PR DESCRIPTION
**Pull request by andylokandy**
Signed-off-by: Andy Lok <andylokandy@hotmail.com>

cherry-picked from https://github.com/tikv/tikv/pull/10794

### What is changed and how it works?

1. Allow getting region info for all regions from the remote store:

`tikv-ctl --host 127.0.0.1:20160 raft region --all-regions`

2. Print region start key and end key with:

`tikv-ctl --host 127.0.0.1:20160 region-properties -r 1`

3. Added subcommand:

- `tikv-ctl unsafe-recover remove-failed-stores -s 1,2,3 -r 1,2,3 --promote-learner`: Promote the learner to leader if only one learner exists
- `tikv-ctl unsafe-recover drop-unapplied-raftlog -r 1,2,3`: regress the commit index to the apply index for the region

### Related changes

- https://github.com/pingcap/kvproto/pull/805
- PR to update `pingcap/docs`/`pingcap/docs-cn`: yes

### Check List 

Tests 

### Release note 

```release-note
1. Allow getting region info for all regions on the remote store:

`tikv-ctl --host 127.0.0.1:20160 raft region --all-regions`

2. Print region start key and end key with:

`tikv-ctl --host 127.0.0.1:20160 region-properties -r 1`

3. Added subcommand:

- `tikv-ctl unsafe-recover remove-failed-stores -s 1,2,3 -r 1,2,3 --promote-learner`: Promote the learner to leader if only one learner exists
- `tikv-ctl unsafe-recover drop-unapplied-raftlog -r 1,2,3`: regress the commit index to the apply index for the region
```

### CODE REVIEW

**Comment by andylokandy** [comment]
@Connor1996 @Little-Wallace PTAL

File: Cargo.toml [hunk 1]
```diff
@@ -238,6 +238,12 @@ test_util = { path = "components/test_util", default-features = false }
 tokio = { version = "1.5", features = ["macros", "rt-multi-thread", "time"] }
 zipf = "6.1.0"
 
+# When you modify TiKV cooperatively with kvproto, this will be useful to submit the PR to TiKV and the PR to
```
Comment (Connor1996) [inline Cargo.toml:241]: no need to change

File: scripts/check-redact-log [hunk 2]
```diff
@@ -7,12 +7,12 @@ function error_msg() {
 }
 
 if [[ "$(uname)" == "Darwin" ]] ; then
-   if grep -r -n --color=always --include '*.rs' --exclude hex.rs --exclude-dir target 'encode_upper' . | grep -v log_wrappers ; then
+   if grep -r -n --color=always --include '*.rs' --exclude hex.rs --exclude-dir tikv-ctl --exclude-dir target 'encode_upper' . | grep -v log_wrappers ; then
```
Comment (Connor1996) [inline scripts/check-redact-log:10]: no need to change?

**Review by Connor1996 — COMMENTED** [review]
LGTM. Though we'd better merge it after verifying the whole process can work correctly.

File: Cargo.toml (hunk 1 above)
Comment (andylokandy) [reply Cargo.toml:241]: Oh, this is rewritten by cherry-pick.

File: scripts/check-redact-log (hunk 2 above)
Comment (andylokandy) [reply scripts/check-redact-log:10]: `make clippy` will complain if this is not changed.

**Review by iosmanthus — APPROVED** [review]
LGTM

**Review by Connor1996 — COMMENTED** [review]
Please update the user doc

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**